File: src/main/java/at/htl/entity/ChosenOption.java
package at.htl.entity;

import at.htl.entity.Answer;
import at.htl.entity.AnswerOption;

import javax.persistence.*;

@NamedQueries({
        @NamedQuery(
                name = "ChosenOption.findAll",
                query = "select co from ChosenOption co"
        )
})
@Entity
@Table(name = "ChosenOption")
public class ChosenOption {
    @EmbeddedId
    ChosenOptionId id;

    public ChosenOption() {
    }

    public ChosenOption(AnswerOption answerOption, Answer answer) {

        id.setAnswerOption(answerOption);
        id.setAnswer(answer);
    }

}

File: leo-survey/leo-survey-frontend/src/app/models/chosen-option.ts
import { Answer } from "./answer";
import { AnswerOption } from "./answer-option";
import { Question } from "./question";

export class ChosenOption {
    constructor(public id = 0,
        public answerOption: AnswerOption = new AnswerOption(),
        public answer : Answer = new Answer(),
        public question : Question = new Question()){}
}
